### DISCUSSION TITLE
Counting multiple objects and loss function

### DISCUSSION THREAD

**Discussion by asadabbas09** [pyg-team/pytorch_geometric#2836] [Q&A]
I have a 3D Point Cloud Dataset which has 5 different types of objects in it.

I have to count the number of objects, e.g. 

```
chair: 4
tv: 1
sofa: 2
bed: 3
car: 0
```
I have count labels available for the different objects during training.

I know the range of individual object counts (0 - 30) as well.

But I'm not sure how I set up the loss function of this problem ( as classification or regression). 

For me exact count is important, so does setting it up as a regression problem makes sense, where I would have 5 outputs, each containing a count estimate for one type of object.

**Comment by rusty1s** [comment]
I personally would tackle this problem as a regression (per object), but I haven't trained a DNN on such a task yet so I'm not sure if there exists more elegant solutions than:
```python
loss = F.mse_loss(out.clamp(0, 30), y)
```
